src/wp-includes/class-wp-scripts.php

```diff
@@ -438,6 +487,12 @@ public function add_inline_script( $handle, $data, $position = 'after' ) {
 		$script   = (array) $this->get_data( $handle, $position );
 		$script[] = $data;
 
+		// Maintain a list of standalone and non-standalone before/after scripts.
+		$standalone_key      = $standalone ? $position . '-standalone' : $position . '-non-standalone';
+		$standalone_script   = (array) $this->get_data( $handle, $standalone_key );
+		$standalone_script[] = $data;
+		$this->add_data( $handle, $standalone_key, $standalone_script );
```

Comment:
The approach of adding new suffixes to the `$position` looks like a hack to me that is prone to side effects. Whether an inline script is standalone or not has nothing to do with the position, and furthermore these new keys here can easily lead to problems. The approach also makes the code harder to understand.

Something else: I believe we had briefly talked about it at some point before, but I don't think splitting inline scripts for _the same handle_ between what is standalone and what isn't is an error-prone approach. By splitting things, we potentially end up printing the scripts in a different order than they were added, which can cause problems. Also I question there's much value in that granularity.

I would much rather suggest that we only keep track for each script handle's "before" and "after" position as a whole whether all inline scripts in the respective position have opted in to standalone. And only if that's the case we can treat the whole inline script blob as standalone.

For example:
* Keep `$position` untouched, so that it only supports "before" or "after".
* Introduce new identifiers "before-is-standalone" and "after-is-standalone", e.g. so that code can call `get_data( $handle, 'after-is-standalone' )`.
* In the `add_inline_script()` method, if `$standalone` is `false`, set `$position . '-is-standalone'` to `false`. If `$standalone` is `true`, set `$position . '-is-standalone'` to `true`, but only if it wasn't previously set to `false` from another inline script in that position.

This will ensure we handle those two concerns separately, that we still load inline scripts in the right order, and that we only load a group of inline scripts in "standalone" mode if all of them were added with `$standalone = true`.